src/NuGet.Clients/NuGetConsole.Host.PowerShell/PowerShellHost.cs

```diff
@@ -726,6 +710,65 @@ public void SetDefaultProjectIndex(int selectedIndex)
             {
                 _solutionManager.Value.DefaultNuGetProjectName = null;
             }
+
+            StartAsyncDefaultProjectUpdate();
+        }
+
+        private void StartAsyncDefaultProjectUpdate()
+        {
+            Assumes.Present(_solutionManager.Value);
+
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Wouldn't this just "move" the UI delay to a different stack. 
I don't understand how it would fix it.

Reply:
There's a bug here which I've fixed in a subsequent comment.  The issue was that if VS was not foreground the default project would not get updated.  If you still have questions after the next commit, then say so.